### PR TITLE
notmuch-mutt: init at 0.22

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Has been my daily driver for ages but outside the nixpkgs repo. Might as well upstream it in case others need it.
###### Things done
- [X] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peterhoeg** [comment]
Travis fails due to an issue compiling gnupg which I haven't touched.

File: pkgs/applications/networking/mailreaders/notmuch/mutt.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{ stdenv, lib, fetchurl, perl, buildPerlPackage, perlPackages, makeWrapper, coreutils
```
Comment (bennofs) [inline pkgs/applications/networking/mailreaders/notmuch/mutt.nix]: The `buildPerlPackage` argument looks unused to me? Or have I missed something?

Comment (peterhoeg) [inline pkgs/applications/networking/mailreaders/notmuch/mutt.nix]: I used ```buildPerlPackage``` previously for this, but realized it wasn't necessary. Taking it out.

**Comment by bennofs** [comment]
Merged in https://github.com/NixOS/nixpkgs/commit/8c04c3c899cfa08e8a5e8a0e7b6d0b323f756ef6.

**Comment by peterhoeg** [comment]
```fetchurl``` can be dropped too, but you merged just before I pushed to the branch.

**Pull request closed without merging**